zip-0316.rst

```diff
@@ -604,15 +613,108 @@ are reserved to indicate Metadata Items other than Receivers or
+the Zcash chain is greater than this value.
+
+The value of a :math:`\mathtt{0xE1}` item MUST be an unsigned 64-bit integer in
+little-endian order specifying a Unix Epoch Time, hereafter referred to as the
+Address Expiry Time. A Unified Address containing Metadata Typecode
+:math:`\mathtt{0xE1}` MUST be considered expired when the current time is
+after the Address Expiry Time.
+
+A Sender that supports `Revision 1`_ of this specification MUST set
+a non-zero ``nExpiryHeight`` field in transactions that it creates. If the
+``nExpiryHeight`` normally constructed by the Sender would be greater than the
+Address Expiry Height, then the transaction MUST NOT be sent. If only an
+Address Expiry Time is specified, then the Sender MUST choose a value for
+``nExpiryHeight`` such that the transaction will expire no more than 24 hours
+after the current time.
+
+If wallet user attempts to send to an expired address, the error presented to
+the user by the wallet SHOULD include a suggestion that the user should
+attempt to obtain a currently-valid address for the intended recipient.
+
+Address expiration imposes no constraints on the Producer of an address. A
+Producer MAY generate multiple Unified Addresses with the same Receivers but
+different expiration metadata and/or any number of distinct Diversified Unified
+Addresses with the same or different expiry metadata, in any combination.
+
+When deriving a UIVK from a UFVK containing Typecodes :math:`\mathtt{0xE0}`
+and/or :math:`\mathtt{0xE1}`, these Metadata Items MUST be retained unmodified
+in the derived UIVK.
+
+When deriving a Unified Address from a UFVK or UIVK containing a Metadata Item
+having Typecode :math:`\mathtt{0xE0}` the derived Unified Address MUST contain
+a Metadata Item having Typecode :math:`\mathtt{0xE0}` such that the Address
+Expiry Height of the resulting address is less than or equal to the Expiry Height
+of the viewing key. 
+
+When deriving a Unified Address from a UFVK or UIVK containing a Metadata Item
+having Typecode :math:`\mathtt{0xE1}` the derived Unified Address MUST contain
+a Metadata Item having Typecode :math:`\mathtt{0xE1}` such that the Address
+Expiry Time of the resulting address is less than or equal to the Expiry Time
+of the viewing key.
+
+Producers of Diversified Unified Addresses should be aware that the expiration
+metadata could potentially be used to link addresses from the same source.
+Normally, if Diversified Unified Addresses derived from the same UIVK contain
+only Sapling and/or Orchard Receivers and no Metadata Items, they will be
+unlinkable as described in [#protocol-concretediversifyhash]_; this property
+does not hold when Metadata Items are present. It is RECOMMENDED that when
+deriving Unified Addresses from a UFVK or UIVK containing expiry metadata that
+the Expiry Height and Expiry Time of each distinct address vary from one
+another, so as to reduce the likelihood that addresses may be linked via their
+expiry metadata.
```

Comment:
Btw, this could be interpreted this in a silly way such as incrementing the expiry time by one second for each Diversified Unified Address. I guess we have to hope that implementors will think about it more than that.